### PR TITLE
Extract Java Library Plugin from plugins project

### PR DESCRIPTION
**Pull request by tresat**

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — COMMENTED** [review]
LGTM other than the comments. Not approving yet due to the failing tests

File: platforms/jvm/plugins-java-library/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    integTestImplementation(testFixtures(project(":resources-http")))
+
+    integTestDistributionRuntimeOnly(project(":distributions-jvm"))
+}
+
+dependencyAnalysis {
+    issues {
+        onRuntimeOnly {
+            // The plugin will suggest moving this to runtimeOnly, but it is needed during :plugins-java-library:compileJava
```
[the remark this answers is not in the document]
Comment (tresat) [reply platforms/jvm/plugins-java-library/build.gradle.kts:55]: Agreed.  I haven't looked any deeper into it, but it seems like it ought to be possible.

File: platforms/jvm/plugins-java-library/src/test/groovy/org/gradle/api/plugins/JavaLibraryPluginTest.groovy [hunk 2]
```diff
@@ -165,6 +165,7 @@ class JavaLibraryPluginTest extends AbstractProjectBuilderSpec {
         defaultConfig.extendsFrom == toSet(runtimeElements)
     }
 
+    @SuppressWarnings('UnnecessaryQualifiedReference') // Needed to prevent No such property: JavaBasePlugin for class: org.gradle.api.plugins.JavaLibraryPluginTest
```
[the remark this answers is not in the document]
Comment (tresat) [reply platforms/jvm/plugins-java-library/src/test/groovy/org/gradle/api/plugins/JavaLibraryPluginTest.groovy:168]: Whatever this ghost was, it's out of my machine now.  Removed.

**Comment by tresat** [comment]
@bot-gradle test this

**Review by big-guy — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**